stablehlo/tests/infer_stablehlo.mlir

```diff
@@ -275,6 +276,16 @@ func.func @batch_norm_inference(%input: tensor<4x256xf32>, %scale: tensor<256xf3
   func.return %1 : tensor<4x256xindex>
 }
 
+// -----
+
+// CHECK-LABEL: func @triangular_solve
+func.func @triangular_solve(%arg0: tensor<10x5x4x4xf32>, %arg1: tensor<10x5x4x4xf32>) -> tensor<10x5x4x4xindex> {
+  %0 = "stablehlo.triangular_solve"(%arg0, %arg1) {left_side = true, lower = true, transpose_a = #stablehlo<transpose NO_TRANSPOSE>, unit_diagonal = true} : (tensor<10x5x4x4xf32>, tensor<10x5x4x4xf32>) -> tensor<10x5x4x4xf32>
```

Comment:
Do we have code coverage for the branches in `inferReturnTypeComponents`, namely `aType` isn't ranked, and `aType` is ranked but `bType` isn't ranked?

Reply:
Good catch! I think that will hurt coverage, fixed.